server/controllers/messages.coffee

```diff
@@ -185,12 +186,17 @@ contentToBuffer = (req, attachment, callback) ->
     # file in the DS, from a previous save of the draft
     # cache it and pass around
     if attachment.url
-        stream = req.message.getBinary filename, (err) ->
+        fileStream = req.message.getBinary filename, (err) ->
             log.error "Attachment streaming error", err if err
 
-        stream_to_buffer_array stream, (err, buffers) ->
-            return callback err if err
-            callback null, Buffer.concat buffers
+        chunks = []
+        bufferer = new stream.Writable
+        bufferer._write = (chunk, enc, next) ->
+            chunks.push(chunk)
+            next()
+        bufferer.end = ->
+            callback null, Buffer.concat chunks
+        fileStream.pipe bufferer
```

Comment:
We kept the full file in memory before and still do. We need to build the email to send it through SMTP & IMAP. Node-imap support for streams2 is not very good so until we can spend a few days making it solid, it is safer to buffer everything. 

The reason for this change is that jdb-cozy-adapter returned the mikeal/request object which act as a full stream (.on 'data') whereas cozydb return a hackish "LaterStream" which only has the .pipe and .abort method. I figure it was simpler to make this hack and remove it when we do the full streaming mentioned in the first § 

It will be cleaner to have it as a "Bufferer" class in helpers. Do you want me to make this change ?